core/src/main/java/com/oath/oak/OakWBufferImpl.java

```diff
@@ -14,15 +14,15 @@
 public class OakWBufferImpl implements OakWBuffer {
 
     private ByteBuffer bb;
-    private final ValueUtils valueOperator;
 
     OakWBufferImpl(Slice s, ValueUtils valueOperator) {
-        this.bb = s.getByteBuffer();
-        this.valueOperator = valueOperator;
-    }
+        ByteBuffer buff = s.getByteBuffer();
+        int initPos = buff.position();
+        buff.position(initPos + valueOperator.getHeaderSize());
+        this.bb = buff.slice();
 
-    private int valuePosition() {
-        return bb.position() + valueOperator.getHeaderSize();
+        // It is important to return the original buffer to its original state.
```

Comment:
If you never need to do anything with the header of the value, why it is not enough to change the BB position to be after the header for as long as OakWBuffer is in use? For other accesses this thread's block's position will be reset anyway... What do I miss?